addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1134,7 +1134,10 @@ def _create_sub_channel(self, from_message_id=None, name=None):
             message = self.env["mail.message"].search([("id", "=", from_message_id)])
         sub_channel = self.create(
             {
-                "channel_member_ids": [Command.create({"partner_id": self.env.user.partner_id.id})],
+                "channel_member_ids": [
+                    Command.create({"partner_id": partner.id})
+                    for partner in self.channel_member_ids.partner_id | self.env.user.partner_id
+                ],
```

Comment:
This could be inefficient, e.g. if we have a channel with ten thousands of members, creating a sub thread is costly.
Any user with any message can make a sub-thread so it'd be nice for sub-thread creation to stay cheap.
When accessing the thread, we could create channel member on the fly if the user is a member of `parent_channel_id`.

I'm not sure whether the presence of channel member is important for other things, e.g. being notified with `@mention`. But if there's no such problem, would be nice to avoid creating a member record if it's not mandatory (again to save space when we could).